core/client-internal/src/main/java/tachyon/client/file/options/SetAttributeOptions.java

```diff
@@ -110,6 +124,66 @@ public boolean getPersisted() {
   }
 
   /**
+   * @return true if the owner value is set, otherwise false
+   */
+  public boolean hasOwner() {
+    return mOwner != null;
+  }
+
+  /**
+   * @return the owner
+   */
+  public String getOwner() {
+    Preconditions.checkState(hasOwner(), PreconditionMessage.MUST_SET_OWNER);
+    return mOwner;
+  }
+
+  /**
+   * @return true if the group value is set, otherwise false
+   */
+  public boolean hasGroup() {
+    return mGroup != null;
+  }
+
+  /**
+   * @return the group
+   */
+  public String getGroup() {
+    Preconditions.checkState(hasGroup(), PreconditionMessage.MUST_SET_GROUP);
+    return mGroup;
+  }
+
+  /**
+   * @return true if the permission value is set, otherwise false
+   */
+  public boolean hasPermission() {
+    return mPermission != Constants.INVALID_PERMISSION;
+  }
+
+  /**
+   * @return the permission
+   */
+  public short getPermission() {
+    Preconditions.checkState(hasPermission(), PreconditionMessage.MUST_SET_PERMISSION);
+    return mPermission;
+  }
+
+  /**
+   * @return true if the recursive value is set, otherwise false
+   */
+  public boolean hasRecursive() {
+    return mRecursive != null;
+  }
+
+  /**
+   * @return the recursive flag value
```

Comment:
this needs more explanation ... in particular recursive only applies to owner, group, and permission